package/MDAnalysis/lib/formats/libmdaxdr.pyx

```diff
@@ -549,6 +520,166 @@ cdef class TRRFile(_XDRFile):
         has_f = bool(has_prop & HASF)
         return TRRFrame(xyz, velocity, forces, box, step, time, lmbda,
                         has_x, has_v, has_f)
+    
+    def read_direct(self, np.float32_t[:, ::1] positions):
+        """
+        Read next frame in the TRR file with positions read directly into
+        a pre-existing array.
+
+        Parameters
+        ----------
+        positions : np.ndarray
+            positions array to read positions into
+
+        Returns
+        -------
+        frame : libmdaxdr.TRRFrame
+            namedtuple with frame information
+
+        See Also
+        --------
+        TRRFrame
+        XTCFile
+
+        Raises
+        ------
+        IOError
+        """
+        if self.reached_eof:
+            raise EOFError('Reached last frame in TRR, seek to 0')
+        if not self.is_open:
+            raise IOError('No file opened')
+        if self.mode != 'r':
+            raise IOError('File opened in mode: {}. Reading only allow '
+                               'in mode "r"'.format('self.mode'))
+
+        return_code = 1
+        cdef int step = 0
+        cdef int has_prop = 0
+        cdef float time = 0
+        cdef float lmbda = 0
+
+        # Use this instead of memviews here to make sure that references are
+        # counted correctly
+
+        cdef np.npy_intp[2] dim
+        dim[0] = self.n_atoms
+        dim[1] = DIMS
+
+        cdef np.npy_intp[2] unitcell_dim
+        unitcell_dim[0] = DIMS
+        unitcell_dim[1] = DIMS
+
+        cdef np.ndarray[np.float32_t, ndim=2] velocity = np.PyArray_EMPTY(2, dim, np.NPY_FLOAT32, 0)
```

Comment:
We now read positions, velocities and forces in the `read_direct_xvf()` function which is the only one called in TRR.py with the overload of `read_next_frame` but I kept this `read_direct()` method so that the base class implementation in XDR.py could theoretically still function.  I am happy to remove it if we want to YAGNI it away. 

Even if we are only reading one attribute you still need to allocate the memory for the others.

Reply:
FIxed.